### ISSUE TITLE
4.1-Classification

### ISSUE THREAD

**Issue by Pin-Jiun** [Pin-Jiun/Machine-Learing-HYL#13]
如果要做資料 Regression（預測）的話，要給予基本的 Model + Data，並且利用 Loss function（損失函數）來檢查現在的 function 的糟糕程度，並利用 Gradient Descent 更新 Model 來模擬窮舉的概念達到找到最佳函數的效果

分類問題會在什麼時候出現?判斷 yes/no 問題, 分辨不同的動物, 分辨人臉

________________________________
### 以天氣分類問題為例子

假設一天的天氣有分成三種，分別是晴天、陰天、雨天，所以相較於 Regression 輸出是一個實數（預測結果），Classification 的 Output 就會分別是晴天、陰天、雨天三種結果

回顧一下機器學習的流程，給 Model（function set） -> Goodness of function -> Find the best function

那我們說過 Goodness of function 可以利用 Loss function 來檢查當前 function 的錯誤程度，然後在最後輸出 best function
在 Regression loss function 可以直接用 Estimation error（估計誤差），然後更新參數用 Gradient Descent，那分類問題呢?

### Loss Function

在 Regression 中，我們的 Loss function 會是預期的答案減去我們計算出來的答案的差異量，但是 Classification 這邊並沒有答案差異這件事情，要怎麼辦呢?

我們平常怎麼知道自己的判斷力好不好？也就是看我們判斷的成功率

所以我們可以利用不符合的答案數量來當作回答狀況的好壞，所以我們可以改寫我們的 Loss function 成
![image](https://user-images.githubusercontent.com/34474924/212008496-2c96be61-80fc-4d00-a6b6-5354232ee450.png)

這邊很明顯不能用 Gradient Descent 去最佳化我們 Model 的參數，那有沒有別的方法可以解決這邊最佳化的問題嗎?
其實是有的，例如 Perceptron Learning Algorithm（PLA）， Support Vector Machine（SVM） 等等，那直接說明 PLA 或是 SVM 會很不直觀，因此我們換個方式解釋

我們先假設一個情境，今天我們的世界只有晴天 & 雨天兩種天氣，因此我們的工作就是判斷明天是晴天 OR 明天是雨天兩種問題
那我們再來假設一下，我們判斷兩種問題的資料也很簡單，就是溫度&濕度兩個變數(feature)，所以如果要把資料畫成圖, 應該怎麼判斷這些資料誰屬於誰？很簡單，那就是用機率來看

_________________________

我們現在有兩個盒子，盒子一跟盒子二，這兩個盒子都有紅色球球跟綠色球球，今天如果我們拿到了一顆紅色的球球，我們只知道這個紅色球球是從這兩個盒子裡面拿到的，請問他是從盒子一拿出來的機率是多少?從盒子二拿出來的機率是多少?

我要算從第一個盒子取出紅色球的機率，就會是取第一個盒子的機率乘上第一個盒子裡面紅色球被拿到的機率去除與兩個盒子拿到紅色球的機率

![image](https://user-images.githubusercontent.com/34474924/212016659-ce3ba4a5-222f-4691-935b-cb80844286eb.png)

我們假設今天我們把盒子改成類別，然後給定一個 x，這個 x 是我們要分類的對象，也就是要分成 class1 或是 class2，那我們這時候先來算這個 x 出現在 class1 跟 class2 中的機率是多少，也就是我從這兩個 class 中抽取到 x 的機率
![image](https://user-images.githubusercontent.com/34474924/212016784-8ace77e5-011c-4d18-9f1a-8210782acdc3.png)

那這邊我們等於需要去取得四個變數

![image](https://user-images.githubusercontent.com/34474924/212016908-59c2d4ef-4c80-4a9c-a043-f10f4a77e3ed.png)

我們只要有這四個變數，我們就能夠計算出任意一個 x 進來之後，他於兩個 cases 機率了，而現在這個做法叫做一個 Generative Model（生成模型），因為我只要能夠有這個模型，我就能生成所有資料點的 Distribution（分布），來判斷資料狀況。

高斯分布

![image](https://user-images.githubusercontent.com/34474924/212017428-8d32c1da-8f2c-49ec-868a-a519ed30341c.png)

這四個變數要怎麼生成，重點要先把P(C1)&P(C2)兩個機率確認，而這個機率怎麼確認，就是利用高斯分布來確認現在訓練的資料組裡面，的分布機率狀態

所以概念上就是只要能搞好高斯分布(找到好的∑和µ)，就能得到兩群出現的機率，就可以間接得到類別中後驗機率，然後就能夠計算它的機率了

_________________

[https://ithelp.ithome.com.tw/articles/10268944](https://ithelp.ithome.com.tw/articles/10268944)
